src/Http/Http.Extensions/src/JsonOptions.cs

```diff
@@ -21,11 +23,21 @@ public class JsonOptions
         // Because these options are for producing content that is written directly to the request
         // (and not embedded in an HTML page for example), we can use UnsafeRelaxedJsonEscaping.
         Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+
+        // The JsonSerializerOptions.GetTypeInfo method is called directly and needs a defined resolver
+        // setting the default resolver (reflection-based) but the user can overwrite it directly or calling
```

Comment:
Given that this is a global instance, I would recommend make it read-only from the get-go using the `JsonSerializerOptions.MakeReadOnly()` method.

Reply:
I believe that here is too early to do this. Users still have a chance to chance the options at this point

Reply:
This will makes the code susceptible to races, in which multiple threads try to set potentially conflicting configuration to the same singleton.